This is a dataset of used cars from eBay Kleinanzeigen, a classifieds section of the German eBay website.

In [2]:
import pandas as pd
import numpy as np

In [3]:
autos = pd.read_csv('autos.csv', encoding='Latin-1')

In [4]:
autos.head()

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

In [5]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   dateCrawled          50000 non-null  object
 1   name                 50000 non-null  object
 2   seller               50000 non-null  object
 3   offerType            50000 non-null  object
 4   price                50000 non-null  object
 5   abtest               50000 non-null  object
 6   vehicleType          44905 non-null  object
 7   yearOfRegistration   50000 non-null  int64 
 8   gearbox              47320 non-null  object
 9   powerPS              50000 non-null  int64 
 10  model                47242 non-null  object
 11  odometer             50000 non-null  object
 12  monthOfRegistration  50000 non-null  int64 
 13  fuelType             45518 non-null  object
 14  brand                50000 non-null  object
 15  notRepairedDamage    40171 non-null  object
 16  date

There's a few missing/NaN values in columns:vehicleType, gearbox, model, fuelType, notRepairedDamage. Might need further inspection

This data uses camel case instead of normal format. Will change the those with the normal format

In [6]:
list(autos.columns)

['dateCrawled',
 'name',
 'seller',
 'offerType',
 'price',
 'abtest',
 'vehicleType',
 'yearOfRegistration',
 'gearbox',
 'powerPS',
 'model',
 'odometer',
 'monthOfRegistration',
 'fuelType',
 'brand',
 'notRepairedDamage',
 'dateCreated',
 'nrOfPictures',
 'postalCode',
 'lastSeen']

In [7]:
autos.rename(columns={'yearOfRegistration':'registration_year',
              'monthOfRegistration':'registration_month',
              'notRepairedDamage':'unrepaired_damage',
              'dateCreated':'ad_created',
                      'nrOfPictures':'num_ad_pictures'
             },inplace=True)

In [8]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'registration_year', 'gearbox', 'powerPS', 'model',
       'odometer', 'registration_month', 'fuelType', 'brand',
       'unrepaired_damage', 'ad_created', 'num_ad_pictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [9]:
autos.describe(include='all')

dateCrawled         name  seller offerType  price abtest  \
count                 50000        50000   50000     50000  50000  50000   
unique                48213        38754       2         2   2357      2   
top     2016-04-02 15:49:30  Ford_Fiesta  privat   Angebot     $0   test   
freq                      3           78   49999     49999   1421  25756   
mean                    NaN          NaN     NaN       NaN    NaN    NaN   
std                     NaN          NaN     NaN       NaN    NaN    NaN   
min                     NaN          NaN     NaN       NaN    NaN    NaN   
25%                     NaN          NaN     NaN       NaN    NaN    NaN   
50%                     NaN          NaN     NaN       NaN    NaN    NaN   
75%                     NaN          NaN     NaN       NaN    NaN    NaN   
max                     NaN          NaN     NaN       NaN    NaN    NaN   

       vehicleType  registration_year  gearbox       powerPS  model  \
count        44905       50000.000000    47320  50000.000000  47242   
unique           8                NaN        2           NaN    245   
top      limousine                NaN  manuell           NaN   golf   
freq         12859                NaN    36993           NaN   4024   
mean           NaN        2005.073280      NaN    116.355920    NaN   
std            NaN         105.712813      NaN    209.216627    NaN   
min            NaN        1000.000000      NaN      0.000000    NaN   
25%            NaN        1999.000000      NaN     70.000000    NaN   
50%            NaN        2003.000000      NaN    105.000000    NaN   
75%            NaN        2008.000000      NaN    150.000000    NaN   
max            NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuelType       brand unrepaired_damage  \
count       50000        50000.000000    45518       50000             40171   
unique         13                 NaN        7          40                 2   
top     150,000km                 NaN   benzin  volkswagen              nein   
freq        32424                 NaN    30107       10687             35232   
mean          NaN            5.723360      NaN         NaN               NaN   
std           NaN            3.711984      NaN         NaN               NaN   
min           NaN            0.000000      NaN         NaN               NaN   
25%           NaN            3.000000      NaN         NaN               NaN   
50%           NaN            6.000000      NaN         NaN               NaN   
75%           NaN            9.000000      NaN         NaN               NaN   
max           NaN           12.000000      NaN         NaN               NaN   

                 ad_created  num_ad_pictures    postalCode  \
count                 50000          50000.0  50000.000000   
unique                   76              NaN           NaN   
top     2016-04-03 00:00:00              NaN           NaN   
freq                   1946              NaN           NaN   
mean                    NaN              0.0  50813.627300   
std                     NaN              0.0  25779.747957   
min                     NaN              0.0   1067.000000   
25%                     NaN              0.0  30451.000000   
50%                     NaN              0.0  49577.000000   
75%                     NaN              0.0  71540.000000   
max                     NaN              0.0  99998.000000   

                   lastSeen  
count                 50000  
unique                39481  
top     2016-04-07 06:17:27  
freq                      8  
mean                    NaN  
std                     NaN  
min                     NaN  
25%                     NaN  
50%                     NaN  
75%                     NaN  
max                     NaN

In [10]:
autos['seller'].value_counts()

privat        49999
gewerblich        1
Name: seller, dtype: int64

In [11]:
autos['offerType'].value_counts()

Angebot    49999
Gesuch         1
Name: offerType, dtype: int64

In [12]:
autos['num_ad_pictures'].value_counts()

0    50000
Name: num_ad_pictures, dtype: int64

In [13]:
autos.drop(['seller','offerType','num_ad_pictures'],axis=1)

dateCrawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
...                    ...                                                ...   
49995  2016-03-27 14:38:19   Audi_Q5_3.0_TDI_qu._S_tr.__Navi__Panorama__Xenon   
49996  2016-03-28 10:50:25  Opel_Astra_F_Cabrio_Bertone_Edition___TÜV_neu+...   
49997  2016-04-02 14:44:48                     Fiat_500_C_1.2_Dualogic_Lounge   
49998  2016-03-08 19:25:42                 Audi_A3_2.0_TDI_Sportback_Ambition   
49999  2016-03-14 00:42:12                                Opel_Vectra_1.6_16V   

         price   abtest vehicleType  registration_year    gearbox  powerPS  \
0       $5,000  control         bus               2004    manuell      158   
1       $8,500  control   limousine               1997  automatik      286   
2       $8,990     test   limousine               2009    manuell      102   
3       $4,350  control  kleinwagen               2007  automatik       71   
4       $1,350     test       kombi               2003    manuell        0   
...        ...      ...         ...                ...        ...      ...   
49995  $24,900  control   limousine               2011  automatik      239   
49996   $1,980  control      cabrio               1996    manuell       75   
49997  $13,200     test      cabrio               2014  automatik       69   
49998  $22,900  control       kombi               2013    manuell      150   
49999   $1,250  control   limousine               1996    manuell      101   

        model   odometer  registration_month fuelType       brand  \
0      andere  150,000km                   3      lpg     peugeot   
1         7er  150,000km                   6   benzin         bmw   
2        golf   70,000km                   7   benzin  volkswagen   
3      fortwo   70,000km                   6   benzin       smart   
4       focus  150,000km                   7   benzin        ford   
...       ...        ...                 ...      ...         ...   
49995      q5  100,000km                   1   diesel        audi   
49996   astra  150,000km                   5   benzin        opel   
49997     500    5,000km                  11   benzin        fiat   
49998      a3   40,000km                  11   diesel        audi   
49999  vectra  150,000km                   1   benzin        opel   

      unrepaired_damage           ad_created  postalCode             lastSeen  
0                  nein  2016-03-26 00:00:00       79588  2016-04-06 06:45:54  
1                  nein  2016-04-04 00:00:00       71034  2016-04-06 14:45:08  
2                  nein  2016-03-26 00:00:00       35394  2016-04-06 20:15:37  
3                  nein  2016-03-12 00:00:00       33729  2016-03-15 03:16:28  
4                  nein  2016-04-01 00:00:00       39218  2016-04-01 14:38:50  
...                 ...                  ...         ...                  ...  
49995              nein  2016-03-27 00:00:00       82131  2016-04-01 13:47:40  
49996              nein  2016-03-28 00:00:00       44807  2016-04-02 14:18:02  
49997              nein  2016-04-02 00:00:00       73430  2016-04-04 11:47:27  
49998              nein  2016-03-08 00:00:00       35683  2016-04-05 16:45:07  
49999              nein  2016-03-13 00:00:00       45897  2016-04-06 21:18:48  

[50000 rows x 17 columns]

seller, num_ad_pictures and offerType columns have almost the same data, can be dropped

In [14]:
autos['abtest'].value_counts()

test       25756
control    24244
Name: abtest, dtype: int64

In [15]:
autos['gearbox'].value_counts()

manuell      36993
automatik    10327
Name: gearbox, dtype: int64

In [16]:
autos['unrepaired_damage'].value_counts()

nein    35232
ja       4939
Name: unrepaired_damage, dtype: int64

In [17]:
autos['odometer'].value_counts()

150,000km    32424
125,000km     5170
100,000km     2169
90,000km      1757
80,000km      1436
70,000km      1230
60,000km      1164
50,000km      1027
5,000km        967
40,000km       819
30,000km       789
20,000km       784
10,000km       264
Name: odometer, dtype: int64

In [18]:
autos['price'].value_counts()

$0         1421
$500        781
$1,500      734
$2,500      643
$1,000      639
           ... 
$64,600       1
$3,012        1
$19,399       1
$3,965        1
$6,889        1
Name: price, Length: 2357, dtype: int64

I noticed that price and odometer, although are numeric data, is non-numeric in autos data. Need to convert it

In [19]:
#odometer has km and coma(,) in it, need to drop it before changing to int
autos['odometer']=autos['odometer'].str.replace('km','').str.replace(',','').astype(int)

In [20]:
autos['price']=autos['price'].str.replace('$','').str.replace(',','').astype(int)

In [21]:
autos.rename(columns={'odometer':'odometer_km'},inplace=True)

In [22]:
autos.head()

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType  price   abtest vehicleType  registration_year    gearbox  \
0  privat   Angebot   5000  control         bus               2004    manuell   
1  privat   Angebot   8500  control   limousine               1997  automatik   
2  privat   Angebot   8990     test   limousine               2009    manuell   
3  privat   Angebot   4350  control  kleinwagen               2007  automatik   
4  privat   Angebot   1350     test       kombi               2003    manuell   

   powerPS   model  odometer_km  registration_month fuelType       brand  \
0      158  andere       150000                   3      lpg     peugeot   
1      286     7er       150000                   6   benzin         bmw   
2      102    golf        70000                   7   benzin  volkswagen   
3       71  fortwo        70000                   6   benzin       smart   
4        0   focus       150000                   7   benzin        ford   

  unrepaired_damage           ad_created  num_ad_pictures  postalCode  \
0              nein  2016-03-26 00:00:00                0       79588   
1              nein  2016-04-04 00:00:00                0       71034   
2              nein  2016-03-26 00:00:00                0       35394   
3              nein  2016-03-12 00:00:00                0       33729   
4              nein  2016-04-01 00:00:00                0       39218   

              lastSeen  
0  2016-04-06 06:45:54  
1  2016-04-06 14:45:08  
2  2016-04-06 20:15:37  
3  2016-03-15 03:16:28  
4  2016-04-01 14:38:50

I will be exploring more on the odometer and price data

In [23]:
autos['odometer_km'].describe()

count     50000.000000
mean     125732.700000
std       40042.211706
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

In [24]:
autos['odometer_km'].value_counts().head()

150000    32424
125000     5170
100000     2169
90000      1757
80000      1436
Name: odometer_km, dtype: int64

In [25]:
autos.loc[autos['odometer_km']>1000000,'name']

Series([], Name: name, dtype: object)

there's 1421 cars that travelled no distance & 11 had over 1million km of mileage

In [26]:
autos['price'].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64

In [27]:
autos['price'].value_counts().head()

0       1421
500      781
1500     734
2500     643
1000     639
Name: price, dtype: int64

there's also 1421 cars that are priced at 0(free)

In [28]:
autos.loc[autos['price']>1000000,'name']

2897      Escort_MK_1_Hundeknochen_zum_umbauen_auf_RS_2000
7814                                           Ferrari_F40
11137    suche_maserati_3200_gt_Zustand_unwichtig_laufe...
22947                           Bmw_530d_zum_ausschlachten
24384                              Schlachte_Golf_3_gt_tdi
27371                                           Fiat_Punto
39377                          Tausche_volvo_v40_gegen_van
39705                          Tausch_gegen_gleichwertiges
42221                                    Leasinguebernahme
47598    Opel_Vectra_B_1_6i_16V_Facelift_Tuning_Showcar...
47634                                          Ferrari_FXX
Name: name, dtype: object

In [29]:
autos.head(30)

dateCrawled                                               name  \
0   2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1   2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2   2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3   2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4   2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
5   2016-03-21 13:47:45  Chrysler_Grand_Voyager_2.8_CRD_Aut.Limited_Sto...   
6   2016-03-20 17:55:21  VW_Golf_III_GT_Special_Electronic_Green_Metall...   
7   2016-03-16 18:55:19                               Golf_IV_1.9_TDI_90PS   
8   2016-03-22 16:51:34                                         Seat_Arosa   
9   2016-03-16 13:47:02          Renault_Megane_Scenic_1.6e_RT_Klimaanlage   
10  2016-03-15 01:41:36                       VW_Golf_Tuning_in_siber/grau   
11  2016-03-16 18:45:34                         Mercedes_A140_Motorschaden   
12  2016-03-31 19:48:22  Smart_smart_fortwo_coupe_softouch_pure_MHD_Pan...   
13  2016-03-23 10:48:32                                 Audi_A3_1.6_tuning   
14  2016-03-23 11:50:46  Renault_Clio_3__Dynamique_1.2__16_V;_viele_Ver...   
15  2016-04-01 12:06:20        Corvette_C3_Coupe_T_Top_Crossfire_Injection   
16  2016-03-16 14:59:02                                Opel_Vectra_B_Kombi   
17  2016-03-29 11:46:22                          Volkswagen_Scirocco_2_G60   
18  2016-03-26 19:57:44               Verkaufen_mein_bmw_e36_320_i_touring   
19  2016-03-17 13:36:21        mazda_tribute_2.0_mit_gas_und_tuev_neu_2018   
20  2016-03-05 19:57:31  Audi_A4_Avant_1.9_TDI_*6_Gang*AHK*Klimatronik*...   
21  2016-03-06 19:07:10                      Porsche_911_Carrera_4S_Cabrio   
22  2016-03-28 20:50:54                               MINI_Cooper_S_Cabrio   
23  2016-03-10 19:55:34   Peugeot_Boxer_2_2_HDi_120_Ps_9_Sitzer_inkl_Klima   
24  2016-04-03 11:57:02                         BMW_535i_xDrive_Sport_Aut.   
25  2016-03-21 21:56:18  Ford_escort_kombi_an_bastler_mit_ghia_ausstattung   
26  2016-04-03 22:46:28                                Volkswagen_Polo_Fox   
27  2016-03-27 18:45:01             Hat_einer_Ahnung_mit_Ford_Galaxy_HILFE   
28  2016-03-19 21:56:19                                      MINI_Cooper_D   
29  2016-04-02 12:45:44  Mercedes_Benz_E_320_T_CDI_Avantgarde_DPF7_Sitz...   

    seller offerType  price   abtest vehicleType  registration_year  \
0   privat   Angebot   5000  control         bus               2004   
1   privat   Angebot   8500  control   limousine               1997   
2   privat   Angebot   8990     test   limousine               2009   
3   privat   Angebot   4350  control  kleinwagen               2007   
4   privat   Angebot   1350     test       kombi               2003   
5   privat   Angebot   7900     test         bus               2006   
6   privat   Angebot    300     test   limousine               1995   
7   privat   Angebot   1990  control   limousine               1998   
8   privat   Angebot    250     test         NaN               2000   
9   privat   Angebot    590  control         bus               1997   
10  privat   Angebot    999     test         NaN               2017   
11  privat   Angebot    350  control         NaN               2000   
12  privat   Angebot   5299  control  kleinwagen               2010   
13  privat   Angebot   1350  control   limousine               1999   
14  privat   Angebot   3999     test  kleinwagen               2007   
15  privat   Angebot  18900     test       coupe               1982   
16  privat   Angebot    350     test       kombi               1999   
17  privat   Angebot   5500     test       coupe               1990   
18  privat   Angebot    300  control         bus               1995   
19  privat   Angebot   4150  control         suv               2004   
20  privat   Angebot   3500     test       kombi               2003   
21  privat   Angebot  41500    

- name: is quite revealing. It shows that many of the items listed are not exactly vehicles. They can be advertisements for vehicle related services, parts of specific vehicles, exchange offers for vehicles, vehicles sold for scrap etc.
- price: is a reflection of the above column. There are prices in the range of $0.00-99.00 which is not usually the price of a vehicle. But these prices were against listings that were set up for exchange offers or toy models of a vehicle. They are not the actual asking price of the vehicle.
- vehicle_type: had missing values in many cases. This was especially true for the cases mentioned above.
registration_year: is a mandatory field. However some of the values in it ranged from 1800-1950 and 2017-5000. This would have to be considered erroneous as the data was scraped in 2016 and as we all know cars were not available in the 1800s
- registration_month: values ranged from 0-12. The listing with 0 against them would have to be considered as erroneous data
brand and model: had missing values. The issue that arose because of this will be revealed later

Therefore, we need to give out charecteristics of a car to filter out data that is not a relevant i.e. car parts, antique cars and valid registration month

we need to set a few criteria for something to be clasified as suitable vehicle data:
- price > 0 
- registration month>0
- registration year>1950

In [30]:
cleaned_autos = autos[autos["price"]!=0]
cleaned_autos = cleaned_autos[cleaned_autos["registration_month"]>0]
cleaned_autos = cleaned_autos[cleaned_autos["registration_year"].between(1949,2018)]
cleaned_autos[:5]

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType  price   abtest vehicleType  registration_year    gearbox  \
0  privat   Angebot   5000  control         bus               2004    manuell   
1  privat   Angebot   8500  control   limousine               1997  automatik   
2  privat   Angebot   8990     test   limousine               2009    manuell   
3  privat   Angebot   4350  control  kleinwagen               2007  automatik   
4  privat   Angebot   1350     test       kombi               2003    manuell   

   powerPS   model  odometer_km  registration_month fuelType       brand  \
0      158  andere       150000                   3      lpg     peugeot   
1      286     7er       150000                   6   benzin         bmw   
2      102    golf        70000                   7   benzin  volkswagen   
3       71  fortwo        70000                   6   benzin       smart   
4        0   focus       150000                   7   benzin        ford   

  unrepaired_damage           ad_created  num_ad_pictures  postalCode  \
0              nein  2016-03-26 00:00:00                0       79588   
1              nein  2016-04-04 00:00:00                0       71034   
2              nein  2016-03-26 00:00:00                0       35394   
3              nein  2016-03-12 00:00:00                0       33729   
4              nein  2016-04-01 00:00:00                0       39218   

              lastSeen  
0  2016-04-06 06:45:54  
1  2016-04-06 14:45:08  
2  2016-04-06 20:15:37  
3  2016-03-15 03:16:28  
4  2016-04-01 14:38:50

In [36]:
top10brands=autos['brand'].value_counts().head(10)

In [44]:
ka=top10brands.index

In [46]:
brand_mean_price = {}
for a_brand in ka:
    brand_mean_price[a_brand] = round(cleaned_autos[cleaned_autos["brand"] == a_brand]["price"].mean(),2)
        

In [47]:
brand_mean_price

{'volkswagen': 5901.0,
 'opel': 5813.93,
 'bmw': 8836.49,
 'mercedes_benz': 31776.8,
 'audi': 9680.72,
 'ford': 7920.74,
 'renault': 2578.34,
 'peugeot': 3201.61,
 'fiat': 2937.46,
 'seat': 4510.38}

In [48]:
mean_price_df = pd.Series(brand_mean_price).to_frame().reset_index()


In [49]:
mean_price_df

index         0
0     volkswagen   5901.00
1           opel   5813.93
2            bmw   8836.49
3  mercedes_benz  31776.80
4           audi   9680.72
5           ford   7920.74
6        renault   2578.34
7        peugeot   3201.61
8           fiat   2937.46
9           seat   4510.38

In [50]:
mean_price_df.rename(columns = {"index":"brands", 0:"avg_price"},inplace=True)
sorted_price=mean_price_df.sort_values("avg_price",ascending=False)

In [51]:
sorted_price

brands  avg_price
3  mercedes_benz   31776.80
4           audi    9680.72
2            bmw    8836.49
5           ford    7920.74
0     volkswagen    5901.00
1           opel    5813.93
9           seat    4510.38
7        peugeot    3201.61
8           fiat    2937.46
6        renault    2578.34